adapters/lora/src/main/java/org/eclipse/hono/adapter/lora/LoraCommandSubscriptionsHandler.java

```diff
@@ -0,0 +1,147 @@
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+
+/**
+ * Class that holds the Lora adapter command subscriptions, that are shared between multiple {@link LoraProtocolAdapter}
+ * verticles. It also takes care to close the subscriptions for deleted/disabled tenants.
+ */
+public class LoraCommandSubscriptionsHandler {
+
+    private final Vertx vertx;
+    private final Tracer tracer;
+    private final Map<SubscriptionKey, TriTuple<CommandConsumer, LoraProvider, Context>> commandSubscriptions;
+
+    /**
+     * Creates a new Lora command subscription handler.
+     *
+     * @param vertx The Vert.x instance to use.
+     */
+    public LoraCommandSubscriptionsHandler(final Vertx vertx) {
+        this(vertx, null);
+    }
+
+    /**
+     * Creates a new Lora command subscription handler.
+     *
+     * @param vertx The Vert.x instance to use.
+     * @param tracer The tracer instance.
+     */
+    public LoraCommandSubscriptionsHandler(final Vertx vertx, final Tracer tracer) {
+        this.commandSubscriptions = new ConcurrentHashMap<>();
+        this.vertx = vertx;
+        this.tracer = tracer == null ? NoopTracerFactory.create() : tracer;
+    }
+
+    /**
+     * Subscribes this handler class for tenant diasbled/deleted events, in order to close command subscriptions to that
+     * tenant.
+     */
+    public void start() {
+        NotificationEventBusSupport.registerConsumer(vertx, TenantChangeNotification.TYPE,
+                notification -> {
+                    if (LifecycleChange.DELETE.equals(notification.getChange())
+                            || (LifecycleChange.UPDATE.equals(notification.getChange())
+                                    && !notification.isTenantEnabled())) {
+                        handleTenantTimeout(notification.getTenantId());
+                    }
+                });
+    }
+
+    private void handleTenantTimeout(final String tenantId) {
+        commandSubscriptions.entrySet().stream().filter(e -> e.getKey().getTenant().equals(tenantId)).forEach(e -> {
+            e.getValue().three().runOnContext(x -> {
+                final Span span = TracingHelper
+                        .buildSpan(tracer, null, "check command subscriptions on tenant timeout",
+                                getClass().getSimpleName())
+                        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
+                        .start();
+                TracingHelper.setDeviceTags(span, tenantId, null);
+                span.log("tenant disabled or removed, corresponding command consumers will be closed");
+
+                @SuppressWarnings("rawtypes")
+                final List<Future> consumerCloseFutures = new ArrayList<>();
+                final Iterator<Entry<SubscriptionKey, TriTuple<CommandConsumer, LoraProvider, Context>>> iterator = commandSubscriptions
+                        .entrySet().iterator();
+                for (final var iter = iterator; iter.hasNext();) {
```

Comment:
`iter` variable is redundant here - you could just use `while (iterator.hasNext()) {`.